### ISSUE TITLE
customEnvVar doesn't work without annotations/descriptors

### ISSUE THREAD

**Issue by sbose78** [redhat-developer/service-binding-operator#368]
If a backing service's CRD/CSV doesn't have any binding metadata, then the variables do not get populated ( shows up a "no value" ).

On adding at least one annotation, the variables magically show up. Even if there is an annotation for `spec.dbName` and nothing for `spec.X.Y`, `spec.X.Y` also works: 
```
servicebindingoperator.redhat.io/spec.dbName: 'binding:env:attribute'
```

Ideally, customEnvVar should work even if no metadata/annotations are present.

**Comment by Avni-Sharma** [comment]
If there is no metadata provided then how would we infer that `A:B` is from a secret object or configmap object?  Providing metadata helps in avoiding collisions in such cases.

**Comment by sbose78** [comment]
We should be able to infer non-secret / non-cm plain attributes or strings at least

**Comment by pedjak** [comment]
We need an acceptance test that can confirm if the issue still exists in the code base. /cc @pmacik

**Comment by pedjak** [comment]
Fixed by #919
